From f26b9c9f9f78903a0a4372ea7749dd08e83e172f Mon Sep 17 00:00:00 2001
From: nana-4 <hnmaigo@gmail.com>
Date: Sun, 1 Dec 2019 13:06:32 +0900
Subject: [PATCH 05/25] Adwaita: Fix some backdrop flat buttons

The commit 6f6070b5 broke the styling.

This fixes it by adding a missing placeholder selector.
---
 gtk/theme/Adwaita/_common.scss           | 1 +
 gtk/theme/Adwaita/gtk-contained-dark.css | 2 +-
 gtk/theme/Adwaita/gtk-contained.css      | 2 +-
 3 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/gtk/theme/Adwaita/_common.scss b/gtk/theme/Adwaita/_common.scss
index c74e5ee316..27cc656bb7 100644
--- a/gtk/theme/Adwaita/_common.scss
+++ b/gtk/theme/Adwaita/_common.scss
@@ -590,6 +590,7 @@ button {
       }
     }
 
+    @at-root %button_basic_flat,
     &.flat { &:backdrop, &:disabled, &:backdrop:disabled { @include button(undecorated); }}
 
     &:disabled {
diff --git a/gtk/theme/Adwaita/gtk-contained-dark.css b/gtk/theme/Adwaita/gtk-contained-dark.css
index f7e8c0ff8c..2d5f375da2 100644
--- a/gtk/theme/Adwaita/gtk-contained-dark.css
+++ b/gtk/theme/Adwaita/gtk-contained-dark.css
@@ -205,7 +205,7 @@ notebook > header > tabs > arrow:backdrop.flat:disabled, button.titlebutton:back
 
 notebook > header > tabs > arrow:backdrop.flat:disabled:active, button.titlebutton:backdrop.flat:disabled:active, notebook > header > tabs > arrow:backdrop.flat:disabled:checked, button.titlebutton:backdrop.flat:disabled:checked, notebook > header > tabs > arrow:backdrop:disabled:active, button.titlebutton:backdrop:disabled:active, notebook > header > tabs > arrow:backdrop:disabled:checked, button.titlebutton:backdrop:disabled:checked, button:backdrop.flat:disabled:active, button:backdrop.flat:disabled:checked, button:backdrop:disabled:active, button:backdrop:disabled:checked { color: #5b5b5b; border-color: #202020; background-image: image(#2a2a2a); box-shadow: inset 0 1px rgba(255, 255, 255, 0); }
 
-notebook > header > tabs > arrow.flat:backdrop, button.flat.titlebutton:backdrop, notebook > header > tabs > arrow.flat:disabled, button.flat.titlebutton:disabled, notebook > header > tabs > arrow.flat:backdrop:disabled, button.flat.titlebutton:backdrop:disabled, button.flat:backdrop, button.flat:disabled, button.flat:backdrop:disabled { border-color: transparent; background-color: transparent; background-image: none; box-shadow: inset 0 1px rgba(255, 255, 255, 0); text-shadow: none; -gtk-icon-shadow: none; }
+notebook > header > tabs > arrow:backdrop, button.sidebar-button:backdrop, button.titlebutton:backdrop, notebook > header > tabs > arrow:disabled, button.sidebar-button:disabled, button.titlebutton:disabled, notebook > header > tabs > arrow:backdrop:disabled, button.sidebar-button:backdrop:disabled, button.titlebutton:backdrop:disabled, notebook > header > tabs > arrow.flat:backdrop, button.flat.titlebutton:backdrop, notebook > header > tabs > arrow.flat:disabled, button.flat.titlebutton:disabled, notebook > header > tabs > arrow.flat:backdrop:disabled, button.flat.titlebutton:backdrop:disabled, button.flat:backdrop, button.flat:disabled, button.flat:backdrop:disabled { border-color: transparent; background-color: transparent; background-image: none; box-shadow: inset 0 1px rgba(255, 255, 255, 0); text-shadow: none; -gtk-icon-shadow: none; }
 
 notebook > header > tabs > arrow:disabled, button.titlebutton:disabled, button:disabled { color: #919190; border-color: #1b1b1b; background-image: image(#323232); text-shadow: none; -gtk-icon-shadow: none; box-shadow: inset 0 1px rgba(255, 255, 255, 0); }
 
diff --git a/gtk/theme/Adwaita/gtk-contained.css b/gtk/theme/Adwaita/gtk-contained.css
index 1310e88c1c..69d0992884 100644
--- a/gtk/theme/Adwaita/gtk-contained.css
+++ b/gtk/theme/Adwaita/gtk-contained.css
@@ -205,7 +205,7 @@ notebook > header > tabs > arrow:backdrop.flat:disabled, button.titlebutton:back
 
 notebook > header > tabs > arrow:backdrop.flat:disabled:active, button.titlebutton:backdrop.flat:disabled:active, notebook > header > tabs > arrow:backdrop.flat:disabled:checked, button.titlebutton:backdrop.flat:disabled:checked, notebook > header > tabs > arrow:backdrop:disabled:active, button.titlebutton:backdrop:disabled:active, notebook > header > tabs > arrow:backdrop:disabled:checked, button.titlebutton:backdrop:disabled:checked, button:backdrop.flat:disabled:active, button:backdrop.flat:disabled:checked, button:backdrop:disabled:active, button:backdrop:disabled:checked { color: #d4cfca; border-color: #d5d0cc; background-image: image(#e4e4e0); box-shadow: inset 0 1px rgba(255, 255, 255, 0); }
 
-notebook > header > tabs > arrow.flat:backdrop, button.flat.titlebutton:backdrop, notebook > header > tabs > arrow.flat:disabled, button.flat.titlebutton:disabled, notebook > header > tabs > arrow.flat:backdrop:disabled, button.flat.titlebutton:backdrop:disabled, button.flat:backdrop, button.flat:disabled, button.flat:backdrop:disabled { border-color: transparent; background-color: transparent; background-image: none; box-shadow: inset 0 1px rgba(255, 255, 255, 0); text-shadow: none; -gtk-icon-shadow: none; }
+notebook > header > tabs > arrow:backdrop, button.sidebar-button:backdrop, button.titlebutton:backdrop, notebook > header > tabs > arrow:disabled, button.sidebar-button:disabled, button.titlebutton:disabled, notebook > header > tabs > arrow:backdrop:disabled, button.sidebar-button:backdrop:disabled, button.titlebutton:backdrop:disabled, notebook > header > tabs > arrow.flat:backdrop, button.flat.titlebutton:backdrop, notebook > header > tabs > arrow.flat:disabled, button.flat.titlebutton:disabled, notebook > header > tabs > arrow.flat:backdrop:disabled, button.flat.titlebutton:backdrop:disabled, button.flat:backdrop, button.flat:disabled, button.flat:backdrop:disabled { border-color: transparent; background-color: transparent; background-image: none; box-shadow: inset 0 1px rgba(255, 255, 255, 0); text-shadow: none; -gtk-icon-shadow: none; }
 
 notebook > header > tabs > arrow:disabled, button.titlebutton:disabled, button:disabled { color: #929595; border-color: #cdc7c2; background-image: image(#faf9f8); text-shadow: none; -gtk-icon-shadow: none; box-shadow: inset 0 1px rgba(255, 255, 255, 0); }
 
-- 
2.20.1

